code/modules/cm_preds/yaut_weapons.dm

```diff
@@ -117,6 +117,209 @@
+				executing = FALSE
+
+		if((INTENT_HARM)) // This is how you farm combo counters, so there's no special interaction.
+			playsound(target, sound_to_play, 50, 1)
+			user.flick_attack_overlay(target, "slam")
+
+
+	if(target != user)
+		if(target.stat == DEAD)
+			return
+		combo_counter++
+		COOLDOWN_START(src, combo_timeout, 15 SECONDS) // This just sets combo to 0 so people dont "pre-stack combos"
+		START_PROCESSING(SSobj, src)
+
+
+/obj/item/weapon/bracer_attachment/chain_gauntlets/process()
+	if(COOLDOWN_FINISHED(src, combo_timeout))
+		combo_counter = 0
+		STOP_PROCESSING(SSobj, src)
+
+/obj/item/weapon/bracer_attachment/chain_gauntlets/proc/get_over_here(mob/target, mob/living/user)
+	var/datum/beam/chain_whip
+	chain_whip = target.beam(target, "chain", 'icons/effects/beam.dmi', 1 SECONDS, beam_type = /obj/effect/ebeam/laser/weak)
+	chain_whip.visuals.alpha = 0
+
+/obj/item/weapon/bracer_attachment/chain_gauntlets/attackby(obj/item/chain_wrapper, mob/user)
+	. = ..()
+	if(istype(chain_wrapper, /obj/item/yautja/chain))
+		if(!has_chain)
+			var/obj/item/yautja/chain/chain_to_wrap = chain_wrapper
+			src.has_chain = TRUE
+			to_chat(user, SPAN_NOTICE("You wrap the [chain_to_wrap] around [src]"))
+			playsound(user, 'sound/weapons/chain_whip.ogg', 50, 1)
+			qdel(chain_to_wrap)
+		else
+			to_chat(user, SPAN_NOTICE("This one already has chains on it!"))
+			return
+
+
+/obj/item/weapon/bracer_attachment/chain_gauntlets/unique_action(mob/user)
+	. = ..()
+	var/mob/living/carbon/human/yautja_user = user
+	if(!gauntlet_deployed)
+		gauntlet_deployed = TRUE
+		punch_knockback = 7
+		yautja_user.start_stomping()
+		addtimer(CALLBACK(src, PROC_REF(undepoy_gauntlets)), 10 SECONDS)
+		yautja_user.visible_message(SPAN_WARNING("[yautja_user] raises the gauntlets infront of its face and starts sprinting!"))
+		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), yautja_user, SPAN_WARNING("You stop covering your face and stop sprinting.")), 10 SECONDS)
+
+	if(gauntlet_deployed)
+		to_chat(user, SPAN_WARNING("You're already charging."))
+		return
+
+/obj/item/weapon/bracer_attachment/chain_gauntlets/proc/undepoy_gauntlets()
+	src.gauntlet_deployed = FALSE
+	punch_knockback = 5
+
+/mob/living/carbon/human/proc/start_stomping(mob/user)
+	src.AddComponent(/datum/component/footstep, 4, 25, 11, 2, "alien_footstep_medium")
+	addtimer(CALLBACK(src, PROC_REF(stop_stomping)), 10 SECONDS)
+	src.species.slowdown += -0.5
+
+/mob/living/carbon/human/proc/stop_stomping(mob/user, obj/item/weapon/bracer_attachment/chain_gauntlets/yautja_glove)
+	src.GetExactComponent(/datum/component/footstep).RemoveComponent()
+	src.species.slowdown -= -0.5
+
+/obj/item/weapon/bracer_attachment/chain_gauntlets/verb/gauntlet_guard()
+	set category = "Weapons"
+	set name = "Guard Yourself"
+	set desc = "Get into a protective stance with your gloves."
+	set src = usr.contents
+
+	unique_action(usr)
```

Comment:
why does this need to be a separate keybind if it just does unique_action, which is already keybindable?